modules/actions/pool/update-liquidity.ts

```diff
@@ -1,33 +1,147 @@
+import vaultV2 from '../../pool/abi/Vault.json';
+import vaultV3 from '../../sources/contracts/abis/VaultV3';
+
+// V2 Vault ABI - only the functions we need
+const v2VaultAbi = vaultV2.filter((item) => item.type === 'function' && item.name === 'getPoolTokens');
+
+// V3 Vault ABI - only the functions we need
+const v3VaultAbi = vaultV3.filter((item) => item.type === 'function' && item.name === 'getPoolTokenInfo');
 
 /**
- * Updates the total liquidity 24h ago for the given pools
+ * Updates the total liquidity 24h ago for the given pools by fetching balances directly from vault contracts
  * Comment: is this really necessary to have in the pools? We have snapshots
  *
- * @param ids
- * @param subgraphClient
- * @param blocksClient
- * @param chain
- * @returns
+ * @param ids Pool IDs to update
+ * @param chain Chain to operate on
+ * @returns Updated pool IDs
  */
-export const updateLiquidity24hAgo = async (
-    ids: string[],
-    subgraphClient: V2SubgraphClient | V3VaultSubgraphClient,
-    chain: Chain,
-) => {
-    // Get liquidity data
+export const updateLiquidity24hAgo = async (ids: string[], chain: Chain, client: ViemClient) => {
+    if (ids.length === 0) return [];
+
+    // Get timestamp and block number
     const ts = chain === Chain.SEPOLIA ? hoursAgo(1) : daysAgo(1);
-    const data = await getLiquidityAndSharesAtTimestamp(chain, ids, subgraphClient, ts);
-    if (!data) return;
+    const blockNumber = await blockNumbers().getBlock(chain, ts);
+
+    if (!blockNumber) {
+        console.log('No block found for timestamp', ts, chain);
+        return [];
+    }
+
+    // Get vault addresses
+    const v2VaultAddress = config[chain].balancer.v2.vaultAddress as `0x${string}`;
+    const v3VaultAddress = config[chain].balancer.v3.vaultAddress as `0x${string}`;
+
+    // Prepare multicall calls
+    const calls = ids.flatMap((id) => [
+        id.length === 42
+            ? {
+                  path: `${id}.poolTokens`,
+                  address: v3VaultAddress,
+                  abi: v3VaultAbi,
+                  functionName: 'getPoolTokenInfo',
+                  args: [id],
+                  parser: (info: any) =>
+                      new Map(info[0].map((token: string, idx: number) => [token.toLowerCase(), info[2][idx]])),
+              }
+            : {
+                  path: `${id}.poolTokens`,
+                  address: v2VaultAddress,
+                  abi: v2VaultAbi as Abi,
+                  functionName: 'getPoolTokens',
+                  args: [id],
+                  parser: (info: any) =>
+                      new Map(info[0].map((token: string, idx: number) => [token.toLowerCase(), info[1][idx]])),
+              },
+        {
+            path: `${id}.totalSupply`,
+            address: id.substring(0, 42) as `0x${string}`,
+            abi: erc20Abi,
+            functionName: 'totalSupply',
+            parser: (totalSupply: bigint) => formatUnits(totalSupply, 18),
+        },
```

Comment:
that wont work because of preminted BPTs in v2. Do we need totalShares at all?

Check if the codebase makes use of `totalShares24hAgo` otherwise we can also drop it

Reply:
it's used in DatastudioService only - is that beets reporting dashboard?